module-cellular/modem/ATCommon.cpp

```diff
@@ -156,3 +162,108 @@ Result Channel::checkResult(bsp::cellular::CellularResultCode cellularResult)
 
     return result;
 }
+
+std::string Channel::anonymizeIfNecessary(std::string text)
+{
+    std::string substringCPIN = "CPIN";
+    std::string substringCLCC = "CLCC";
+    std::string substringATD  = "ATD";
+    std::string substringCPBR = "CPBR";
+    std::string substringCSPN = "CSPN";
+
+    if (text.find(substringCPIN) != std::string::npos) {
+        text = anonymizeInQuotationMarks(text);
+    }
+    else if (text.find(substringCLCC) != std::string::npos) {
+        text = anonymizeInQuotationMarks(text, 2);
+    }
+    else if (text.find(substringATD) != std::string::npos) {
+        text = anonymizeNumbers(text, 2);
+    }
+    else if (text.find(substringCPBR) != std::string::npos) {
+        text = anonymizeInQuotationMarks(text, 1);
+    }
+    else if (text.find(substringCSPN) != std::string::npos) {
+        text = anonymizeInQuotationMarks(text, 2);
+    }
+
+    return text;
```

Comment:
I have also another proposition for this function, IMO it looks better than this else if block :) 

```std::string anonymizeIfNecessary(const std::string& text){
    std::map <std::string, std::function<std::string()>> anonymizeOption
    {
    {substringCPIN,[&text](){return anonymizeInQuotationMarks (text);}},
    {substringCLCC,[&text](){return anonymizeInQuotationMarks (text, 2);}},
    {substringATD,[&text](){return anonymizeNumbers (text, 2);}},
    {substringCPBR,[&text](){return anonymizeInQuotationMarks (text, 1);}},
    {substringCSPN,[&text](){return anonymizeInQuotationMarks (text, 2);}}
    };
    
    for(const auto& [key, anonymizeFunction] : anonymizeOption){
        if(text.find(key) != std::string::npos){
            return anonymizeFunction();
        }
    }
    return text;
    
}```